### ISSUE TITLE
App status indicator

### ISSUE THREAD

**Issue by sguergachi** [yyoon/Journaley#43]
For use in possibly long process and for visual indication of background work, like auto saving and editing entries, loading of entries ect.

There are 3 styles I like, we can choose whichever makes the most sense to the user without having to explicitly point out this is a progress indicator.

_Note_ the timing on the 3rd one is off, it would be much slower to show changing states.

**1**
![animate](https://cloud.githubusercontent.com/assets/615386/3336978/2d00a716-f841-11e3-98ca-16045a49a528.gif)
**2**
![animate2](https://cloud.githubusercontent.com/assets/615386/3336980/32efc134-f841-11e3-8d91-2dfe51587596.gif)
**3**
![animate3](https://cloud.githubusercontent.com/assets/615386/3336982/36867e32-f841-11e3-96a4-27dd62a5e455.gif)

**Comment by sguergachi** [comment]
Thinking about this again. Showing "saving.." every 3 seconds is not a good idea and will become annoying. I think using a subtle shape as an indicator is a better idea, but this time it shouldn't be flashing. When I come up with something I will sketch it up.

**Comment by yyoon** [comment]
What scenarios are you thinking of displaying the progress indicator?

In terms of implementation, showing a progress indicator takes a lot of work. Let's take the entry saving , for example. Currently the saving happens with one line of code, which returns after saving is complete. There are no intermediate steps that I can control. In order to display a progress indicator, I would have to change that to some more complicated code that works asynchronously, and calling a callback method at intermediate steps (which would allow us to do something to update the progress indicator), and then finally complete. This even assumes that such method exists in the library, which may not be the case.

On the other hand, for *loading* all the entries, it may be a bit easier because the loading logic is fully handled by our code, and we could update the progress indicator every time a new entry is loaded into memory, for example.

So, I would suggest first thinking about the exact scenarios where we want to display the progress indicator, and then move forward.

**Comment by sguergachi** [comment]
I don't think we need to be accurate in how long progress is taking. It should just indicate that progress has occurred. So for instance, we can just indicate that saving has occurred after the fact, no need to show that saving is occurring and then has stopped occurring. Something similar to how Word or IDE's show a '*' next to the file name to indicate a changed file hasn't been saved, except we wouldn't tell the user that the entry *hasn't* been saved but when it *has*.

Again, still need to think specifically how this is going to be implemented. But the basic idea is to reassure users by giving some surface level insight into what the app has done for them in the background.

Not sure if this makes sense. 

-Sammy Guergachi
On 7/16/2015 1:02:18 AM, YoungSeok Yoon <notifications@github.com> wrote:
What scenarios are you thinking of displaying the progress indicator?
In terms of implementation, showing a progress indicator takes a lot of work. Let's take the entry saving , for example. Currently the saving happens with one line of code, which returns after saving is complete. There are no intermediate steps that I can control. In order to display a progress indicator, I would have to change that to some more complicated code that works asynchronously, and calling a callback method at intermediate steps (which would allow us to do something to update the progress indicator), and then finally complete. This even assumes that such method exists in the library, which may not be the case.
On the other hand, for loading all the entries, it may be a bit easier because the loading logic is fully handled by our code, and we could update the progress indicator every time a new entry is loaded into memory, for example.
So, I would suggest first thinking about the exact scenarios where we want to display the progress indicator, and then move forward.

**Comment by yyoon** [comment]
I guess the word "progress" is somewhat bugging me, but I agree that it's good to indicate the user that the app is doing something useful in the background. Let's discuss this later when you come up with a little more polished idea.
